lua/nvim-tree/watcher.lua

```diff
@@ -7,49 +7,62 @@ local M = {}
 local Watcher = {}
 Watcher.__index = Watcher
 
+-- TODO check for a Watcher with the same opts.absolute_path and return that
 function Watcher.new(opts)
   if not M.enabled then
     return nil
   end
   log.line("watcher", "Watcher:new   '%s'", opts.absolute_path)
 
-  local ok, fs_event = pcall(uv.new_fs_event)
-  if not ok then
-    utils.warn(string.format("Could not initialize a watcher for path %s", opts.absolute_path))
+  local stat, _ = uv.fs_stat(opts.absolute_path)
+  if not stat or stat.type ~= "directory" then
     return nil
   end
 
   local watcher = setmetatable({
-    _path = opts.absolute_path,
-    _w = fs_event,
+    _opts = opts,
   }, Watcher)
 
-  return watcher:start(opts)
+  return watcher:start()
 end
 
-function Watcher:start(opts)
-  log.line("watcher", "Watcher:start '%s'", self._path)
-  local ok = pcall(
-    uv.fs_event_start,
-    self._w,
-    self._path,
-    {},
-    vim.schedule_wrap(function()
-      opts.on_event(self._path)
-    end)
-  )
-  if not ok then
-    utils.warn(string.format("Could not start the watcher for path %s", self._path))
+function Watcher:start()
+  log.line("watcher", "Watcher:start '%s'", self._opts.absolute_path)
+
+  local rc, name
+
+  self._p, _, name = uv.new_fs_poll()
```

Comment:
The [fail](https://github.com/luvit/luv/blob/master/docs.md#error-handling) pseudo-type makes me sad.